gpflow/conditionals/util.py

```diff
@@ -51,6 +51,18 @@ def base_conditional(Kmn: tf.Tensor,
         0)  # [N]
     Kmn = tf.transpose(Kmn, perm)  # [..., M, N]
```

Comment:
We can't have the assert_shapes check at the start of the function body as the function signature expects Kmn with shape [M, ..., N], but assert_shapes() only handles *leading* Ellipsis dimensions - so we need to add the shape constraint *after* rolling it to [..., M, N]. Would it not be cleaner to simply expect the input to be [..., M, N] in the first place? or if we want the ... to be next to the N, change it to take `Knm` instead of `Kmn` and expect shape [..., N, M]?

Reply:
seeing as its not a critical change I wouldn't change this important interface at this point for a mildly aesthetic improvement.

Reply:
I'm happy to defer that to another PR but I think it's an important clarity-of-design issue - "leading" dimensions in the middle seems rather confusing!

Reply:
Moved into new issue: #1296